src/main/java/com/fasterxml/jackson/databind/deser/std/StdKeyDeserializer.java

```diff
@@ -417,6 +431,18 @@ private EnumResolver _getToStringResolver(DeserializationContext ctxt)
             }
             return res;
         }
+
+       private EnumResolver _getIndexResolver(DeserializationContext ctxt) {
+          EnumResolver res = _byIndexResolver;
```

Comment:
this form of double-checked locking is not thread-safe, see https://shipilev.net/blog/2014/safe-public-construction/

_byIndexResolver needs to be volatile